pkg/detector/detector.go

```diff
@@ -777,23 +793,21 @@ func (d *ResourceDetector) ClaimClusterPolicyForObject(object *unstructured.Unst
 
 	objectCopy := object.DeepCopy()
 
-	util.MergeLabel(objectCopy, policyv1alpha1.ClusterPropagationPolicyLabel, policy.Name)
 	util.MergeLabel(objectCopy, policyv1alpha1.ClusterPropagationPolicyPermanentIDLabel, policyID)
-
-	util.MergeAnnotation(objectCopy, policyv1alpha1.ClusterPropagationPolicyAnnotation, policy.Name)
+	util.ReplaceAnnotation(objectCopy, policyv1alpha1.ClusterPropagationPolicyAnnotation, policy.Name)
```

Comment:
Can you check if it has something to do with this in #4743? The previous CI error was due to the incorrect use of the MergeAnnotation function.

Reply:
cc @whitewindmills

Reply:
I've made a separate PR #4751 to fix this.